app/services/spine_service/reports/clinic/diagnosis_report.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module SpineService
+  module Reports
+    module Clinic
+      class DiagnosisReport
+        include ModelUtils
+        attr_reader :start_date, :end_date
+
+        def initialize(start_date:, end_date:, **kwargs)
+          @start_date = start_date.to_date.beginning_of_day.strftime('%Y-%m-%d %H:%M:%S')
+          @end_date = end_date.to_date.end_of_day.strftime('%Y-%m-%d %H:%M:%S')
+        end
+
+        def fetch_report
+          type = EncounterType.find_by_name 'Outpatient diagnosis'
+          data = Encounter.where('encounter_datetime BETWEEN ? AND ?
+            AND encounter_type = ?
+            AND obs.concept_id IN(6543, 6542)', @start_date, @end_date, type.id).\
+            joins('INNER JOIN obs ON obs.encounter_id = encounter.encounter_id
+            INNER JOIN person p ON p.person_id = encounter.patient_id
+            LEFT JOIN person_name n ON n.person_id = encounter.patient_id AND n.voided = 0
+            LEFT JOIN person_attribute z ON z.person_id = encounter.patient_id AND z.person_attribute_type_id = 12
+            RIGHT JOIN person_address a ON a.person_id = encounter.patient_id
```

Comment:
This is dangerous. The RIGHT joins disregards the everything from the left table. So this will return even address that we are not interested in.

But I'm interested to know why the choice of the RIGHT JOIN